src/extensions/scratch3_vizblocks/index.js

```diff
@@ -136,6 +174,346 @@ class Scratch3VizBlocks {
+
+        const penSkinId = this._getPenLayerID();
+        if (penSkinId >= 0) {
+            const penState = this._getPenState(target);
+            
+            target.x = this._xCenter;
+            target.y = this._yCenter + this._height;
+            // draw y axis line
+            this.runtime.renderer.penLine(penSkinId, penState.penAttributes, this._xCenter, this._yCenter, target.x, target.y);
+            
+            // set text for y axis
+            this.setText(penSkinId, penState, label, 'Y', target);
+            this.runtime.requestRedraw();
+        }
+    }
+
+    /**
+     * Draw line using functions from "pen block"
+     * @param {object} args - the block arguments.
+     * @param {object} util - utility object provided by the runtime.
+     */
+    drawLine (args, util){
+        const dataX = this._xPos;
+        const dataY = this._yPos;
+        
+        const target = util.target;
+        const penSkinId = this._getPenLayerID();
+        if (penSkinId >= 0 && dataX.length > 1 && dataY.length > 1) {
+            const penState = this._getPenState(target);
+            for (let i = 0; i < dataX.length; i++){
+                this.runtime.renderer.penPoint(penSkinId, penState.penAttributes, dataX[i], dataY[i]);
+            }
+            for (let i = 1; i < dataX.length; i++){
+                // eslint-disable-next-line max-len
+                this.runtime.renderer.penLine(penSkinId, penState.penAttributes, dataX[i - 1], dataY[i - 1], dataX[i], dataY[i]);
+            }
+            this.runtime.requestRedraw();
+        }
+    }
+
+    /**
+     * read data from input as (x, y)
+     * @param {object} args - the block arguments.
+     */
+    readXY (args){
+        const x = Cast.toNumber(args.X);
+        const y = Cast.toNumber(args.Y);
+
+        this._xArray.push(x);
+        this._yArray.push(y);
+        this._posEmpty = false;
+    }
+
+    /**
+     * load number costumes from existing assets, add costumes to target of the utility object
+     * @param {RenderedTarget} target - target object that has been updated.
+     * @returns {Promise.<Array>} -  A promise for the requested costumes Array.
+     *   If the promise is resolved with non-null, the value is the requested asset or a fallback.
+     *   If the promise is resolved with null, the desired asset could not be found with the current asset sources.
+     *   If the promise is rejected, there was an error on at least one asset source. HTTP 404 does not count as an
+     *   error here, but (for example) HTTP 403 does.
+     */
+    loadCostumes (target){
+        // console.log('draw text');
+        let result = Promise.resolve();
+        const costumes = [];
+
+        // load number costumes and letter costumes
+        assetIDArr.forEach((assetID, index) => {
+            const costumeObject = new Object();
+            const name = index >= 10 ? String.fromCharCode('A'.charCodeAt() + index - 10) : index;
```

Comment:
`+ index - 10` seems like it is not needed when I tried it. Also, we are missing the letter `Z`.

```suggestion
            const name = index >= 10 ? String.fromCharCode('A'.charCodeAt()) : index;
```